Source/Core/InputCommon/ControlReference/ExpressionParser.h

```diff
@@ -140,7 +140,14 @@ class ControlQualifier
 class ControlEnvironment
 {
 public:
-  using VariableContainer = std::map<std::string, ControlState>;
+  struct VariableState
+  {
+    VariableState() : state(0), reference_counter(0) {}
```

Comment:
The member variables can be initialized directly in the struct instead of in the initializer list, which cuts down on code a little bit.